crates/rethnet_evm/src/state/fork.rs

```diff
@@ -203,60 +224,80 @@ impl crate::DatabaseDebug for ForkDatabase {
         index: U256,
         value: U256,
     ) -> Result<(), Self::Error> {
-        self.layered_db
+        self.layered_state
             .set_account_storage_slot(address, index, value)
-            .map_err(ForkDatabaseError::LayeredDatabase)
+    }
+
+    fn set_block_context(
+        &mut self,
+        block_number: U256,
+        state_root: &B256,
+    ) -> Result<(), Self::Error> {
+        self.state_root_to_block_number
+            .insert(*state_root, block_number);
+        self.set_state_root(state_root)?;
+        Ok(())
+    }
+
+    fn restore_fork_block_context(&mut self, state_root: &B256) -> Result<(), Self::Error> {
```

Comment:
Why not just call `set_state_root` instead of adding a new function that does the same?